### PR TITLE
Add more developer documentation

### PR DESCRIPTION
**Pull request by lovettchris**
Splitting it out into separate dev.md file with associated images, etc.

### CODE REVIEW

File: docs/dev.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+To debug the tests use the launch configuration named `Extension Tests - SuiteName`.
+
+**Note:** if breakpoints are not working, try changing one line of code in `function activate` in `extension.ts`, even
+adding a newline seems to work, then press F5.
+
+## Packaging
+
+To publish the extension on the VS Code marketplace you
+should follow [these instructions](https://code.visualstudio.com/api/working-with-extensions/publishing-extension).
```
Comment (gebner) [inline docs/dev.md:36]:
```suggestion
To publish the extension on the VS Code marketplace you
should run the `./release.sh` script.
```

Comment (lovettchris) [reply docs/dev.md]: ok, I merged your link to the script, but included the link for more information (that link shows how to install vsce, etc)

File: README.md [hunk 2]
```diff
@@ -286,45 +286,4 @@ If this version is not yet installed `lean --version` will install it.
...
-should follow [these instructions](https://code.visualstudio.com/api/working-with-extensions/publishing-extension).
-But for a quick summary, run this command:
-```
-cd <vscode-lean4-repo>/vscode-lean4
-npx vsce package
-```
-
-This gives you a .vsix which you can install using the VS Code
-command `Extensions: Install from VSIX...`.
+See also [Development](docs/dev.md).
```
Comment (Vtec234) [inline README.md:289]:
```suggestion
## For VSCode extension developers

See [Development](docs/dev.md).
```
Let's keep the header to make this easier to find.

File: docs/dev.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+## Packaging
+
+To publish the extension on the VS Code marketplace you
+should run the `./release.sh` script.
+
+See [publishing extensions](https://code.visualstudio.com/api/working-with-extensions/publishing-extension)
+for more information.
+
+You can build a local .vsix for testing using `npx vsce package`.
```
Comment (Vtec234) [inline docs/dev.md:41]:
```suggestion
You can build a local .vsix for testing using `npm run package` in the `vscode-lean4/` subdirectory.
```

File: docs/dev.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+command `Extensions: Install from VSIX...`.
+
+## Troubleshooting
+
+**[!] SyntaxError: Cannot use import statement outside a module**
+
+On Windows the build can fail if the path in your Terminal window has the wrong "case" on the drive letter.
+For example, if your terminal was started in `d:\git\leanprover\vscode-lean4` and you get this error then
+try and start your terminal using an upper case driver letter like this: `D:\git\leanprover\vscode-lean4`.
+Then the build should succeeded.  This is an open bug in [bug in rollup](https://github.com/rollup/rollup/issues/4439).
```
Comment (Vtec234) [inline docs/dev.md:53]:
```suggestion
Then the build should succeed.  This is an open [bug in Rollup](https://github.com/rollup/rollup/issues/4439).
```

**Review by Vtec234 — APPROVED** [review]
Thank you!

**Pull request merged**